### PR TITLE
Ek 428 gatsby caching issue

### PR DESCRIPTION
**Pull request by cs-paz**
Installed `@netlify/plugin-gatsby` as an alternative to using gatsby's caching system for the time being due to an error caused by `gatsby-source-filesystem`.  It seems that the gatsby community has only got around this error by running `gatsby clean` at the start of each build which would be the opposite of the purpose of this branch.
See [here](https://stackoverflow.com/questions/56226824/gatsbyjs-invariant-violation-encountered-an-error-trying-to-infer-a-graphql-typ) and [here](https://github.com/LekoArts/gatsby-starter-prismic/issues/19)

Test:

1. Gatsby build time has been notably reduced.
2. Faculty Nodes have consistent ID's between builds
3. Gatsby uses cache instead of fetching image when a photo exists - Note, for this test to pass, we will need to initiate a rebuild without clearing the cache 2-3 times and ensure that when there exists cached images, the number of requests hitting the api should be 0 (unless there was a change for a corresponding node in the api coincidentally happening while we are testing)

Pre deploy: Merge https://github.com/Stevens-EWS/gatsby-source-custom-api/pull/1/files and make a commit changing package.json to reference master instead of the commit hash

### CODE REVIEW

File: gatsby_fp/gatsby-config.js [hunk 1]
```diff
@@ -38,6 +38,8 @@ module.exports = {
             notable_courses: String
             school: String
             ses_department: String
+            alt_publications: String
+            notable_publications: String
```
Comment (dmotan) [inline gatsby_fp/gatsby-config.js:42]: I believe this change is unrelated to the caching issue, if yes, I think this should be in a separate PR.

Comment (cs-paz) [reply gatsby_fp/gatsby-config.js:42]: I think this code has already been merged to master... this branch was probably made before that PR (#50) was merged.  I'll remove this code or manually resolve the merge conflicts once our testing is done so we don't get any unrelated issues during testing caused by removing this.

File: gatsby_fp/package.json [hunk 2]
```diff
@@ -44,6 +44,7 @@
     "shortid": "^2.2.16"
   },
   "devDependencies": {
+    "@netlify/plugin-gatsby": "^1.0.3",
```
Comment (dmotan) [inline gatsby_fp/package.json:47]: So that I am understanding this properly, adding this plugin is sufficient to enable caching?

File: gatsby_fp/gatsby-config.js (hunk 1 above)
Comment (mf-stevens) [reply gatsby_fp/gatsby-config.js:42]: When a code change is already in master but you have some feature branch that was created before that point, and you want the feature branch to also contain that code change, the best way to achieve that is to merge master into the feature branch, not manually commit the same code to the feature branch. If said merge causes conflicts, resolve them during the merge.

When this particular change was committed to this feature branch, the commit message was "Fixed merge conflicts from master" which suggests that it was done properly. On bitbucket, this would not show up as part of the full diff when looking at the feature branch PR, but maybe Github is different.

File: gatsby_fp/package.json (hunk 2 above)
Comment (cs-paz) [reply gatsby_fp/package.json:47]: Yes, just having this plugin installed cuts our netlify build times in half

File: gatsby_fp/gatsby-node.js [hunk 3]
```diff
@@ -38,13 +37,14 @@ exports.onCreateNode = async ({
       // err returns a string value with no ability to
       // get the status code without searching for it within the string
       (async function getStatusCodeUsingFetch() {
-        const response = await fetch(node.photo_url) 
+        console.log(err)
+        const response = await fetch(node.photo_url)
         if(response.status != 404 ){
           throw new Error(`Invalid Status Code [${response.status}] for photo_url of user ${node.pf_username}`)
```
Comment (dmotan) [inline gatsby_fp/gatsby-node.js:42]: I know this should have been caught a long time ago, but we're finding and learning new things as we move forward with this project. I am not sure on what this error message is based on, it displays the same error message with any kind of error that happens in `createRemoteFileNode`, for example, I removed the `getCache` property from the object above, and it threw this error message. When I first saw the error message, I thought this might be related to overloading the People API, but since I have not changed `GATSBY_CONCURRENT_DOWNLOAD` this ensures that it has nothing to do with the People API overloading. After debugging and adding some console logs I found that the error was due to removing `getCache`. We should tweak the error handling here to better describe the error.

**Comment by dmotan** [comment]
There is no need for this PR anymore, the issue with caching is being solved in https://github.com/Stevens-EWS/faculty-profiles/pull/49

**Comment by mf-stevens** [comment]
Closing before merge because this site has been decommissioned.
